base/path.jl

```diff
@@ -211,13 +211,15 @@ julia> splitpath("/home/myuser/example.jl")
 ```
 """
 function splitpath(p::AbstractString)
-    out = ()
+    out = AbstractString[]
```

Comment:
Better to make it just `String`, Postel's Law and all: this should take all kinds of strings but return one very specific kind of string vector, i.e. `Vector{String}`.

Reply:
>  I'm pretty sure `dirname` et al. always return `String` anyway.

```julia
help?> dirname
search: dirname

  dirname(path::AbstractString) -> AbstractString
```
That's actually the reason I made it an `AbstractString`, was to match `dirname`/`basename`.

... But actually those end up calling `splitdir`, which definitely does return a `String`. _And_, despite what the docstring says, it also only accepts a `String`?:
https://github.com/JuliaLang/julia/blob/a2fdb7d078f9281bbaf49133cf37f1344ab76b59/base/path.jl#L131-L137

Reply:
Well, I've changed my `splitpath` to match: Take and return `String`, but say `AbstractString` in the docs.. I want to match the style of the other functions.

If those should be changed, this can be done in the same batch. :)